### PR TITLE
Fixes for Post-Battle Handling of Capital Fighters

### PR DESCRIPTION
**Pull request by mkerensky**
+ Post-battle armor scale conversion for capital fighters
+ Mapping of squadron weapon/equipment critical hits to individual fighters
Fixes bug #587 
Also, allows weapon and BAP/ECM critical hits taken by squadrons to be recorded and repaired in MHQ

### CODE REVIEW

File: megamek/src/megamek/server/Server.java [hunk 1]
```diff
@@ -26831,18 +26847,34 @@ else if (secondaryEffects) {
                     for (Mounted weap : aero.getWeaponList()) {
                         if (weap.getLocation() == loc) {
                             if (destroyAll) {
-                                weap.setHit(true);
+                                weap.setHit(true);
```
Comment (sixlettervariables) [inline megamek/src/megamek/server/Server.java]: Odd whitespace addition

File: megamek/src/megamek/common/Aero.java [hunk 2]
```diff
@@ -4055,7 +4055,7 @@ public boolean canSpot() {
     // StratOps pg. 32 & 34
     @Override
     public void doDisbandDamage() {
-
+
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/Aero.java]: Whitespace addition

**Comment by HammerGS** [comment]
This pull request **fixes 1 alert** when merging d98dd1eacce1a09b156d4b1fdd66ed0b31c9a82a into 96ca7e6769a85324ad1dc9ae15e063387fba916e - [view on LGTM.com](https://lgtm.com/projects/g/MegaMek/megamek/rev/pr-07baa10cee69baa6c045bc22207cf4357744731d)

**fixed alerts:**

* 1 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: megamek/src/megamek/server/Server.java (hunk 1 above)
Comment (mkerensky) [reply megamek/src/megamek/server/Server.java]: Fixed

File: megamek/src/megamek/common/Aero.java (hunk 2 above)
Comment (mkerensky) [reply megamek/src/megamek/common/Aero.java]: Fixed

**Comment by HammerGS** [comment]
This pull request **fixes 1 alert** when merging 479591b8cd67ecdda6869c92d0847796b5a10d33 into 96ca7e6769a85324ad1dc9ae15e063387fba916e - [view on LGTM.com](https://lgtm.com/projects/g/MegaMek/megamek/rev/pr-c0c5f9ca4ed6814f49a19acc65b79b9d4ee456b1)

**fixed alerts:**

* 1 for Dereferenced variable may be null

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**